source/MaterialXRenderGlsl/GLTextureHandler.cpp

```diff
@@ -45,6 +50,14 @@ bool GLTextureHandler::acquireImage(const FilePath& filePath,
                                     bool generateMipMaps,
                                     const Color4* fallbackColor)
 {
+    if(!_boundTextureLocationsInitialized)
```

Comment:
You could just check for _boundTextureLocations size == 0 means to initialize it instead of requiring an extra variable.
Also we should throw an exception if the return value is 0 as it means there are no units at all.